client/trino-client/src/main/java/io/trino/client/JsonCodec.java

```diff
@@ -33,18 +36,29 @@
 public class JsonCodec<T>
 {
     // copy of https://github.com/airlift/airlift/blob/master/json/src/main/java/io/airlift/json/ObjectMapperProvider.java
-    static final Supplier<ObjectMapper> OBJECT_MAPPER_SUPPLIER = () -> new ObjectMapper()
-            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
-            .disable(MapperFeature.AUTO_DETECT_CREATORS)
-            .disable(MapperFeature.AUTO_DETECT_FIELDS)
-            .disable(MapperFeature.AUTO_DETECT_SETTERS)
-            .disable(MapperFeature.AUTO_DETECT_GETTERS)
-            .disable(MapperFeature.AUTO_DETECT_IS_GETTERS)
-            .disable(MapperFeature.USE_GETTERS_AS_SETTERS)
-            .disable(MapperFeature.CAN_OVERRIDE_ACCESS_MODIFIERS)
-            .disable(MapperFeature.INFER_PROPERTY_MUTATORS)
-            .disable(MapperFeature.ALLOW_FINAL_FIELDS_AS_MUTATORS)
-            .registerModule(new Jdk8Module());
+    static final Supplier<ObjectMapper> OBJECT_MAPPER_SUPPLIER = () -> {
+        JsonFactory jsonFactory = JsonFactory.builder()
+                .streamReadConstraints(StreamReadConstraints.builder()
+                    .maxStringLength(Integer.MAX_VALUE)
+                    .maxNestingDepth(Integer.MAX_VALUE)
```

Comment:
This is safe, because we trust the server not to send abusive json, right? Is it worth adding a comment?

Reply:
Yup. This was the default behaviour either way, before we've updated Jackson